addons/adyen_platforms/models/adyen_shareholder.py

```diff
@@ -0,0 +1,126 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import uuid
+
+from odoo import api, fields, models
+
+from odoo.addons.adyen_platforms.models.adyen_kyc import ADYEN_KYC_STATUS
+
+
+class AdyenShareholder(models.Model):
+    _name = 'adyen.shareholder'
+    _inherit = ['adyen.id.mixin', 'adyen.address.mixin']
+    _description = 'Adyen for Platforms Shareholder'
+    _rec_name = 'full_name'
+
+    adyen_account_id = fields.Many2one('adyen.account', ondelete='cascade')
+    shareholder_reference = fields.Char('Reference', default=lambda self: uuid.uuid4().hex)
+    shareholder_uuid = fields.Char('UUID')  # Given by Adyen
+    first_name = fields.Char('First Name', required=True)
```

Comment:
https://www.w3.org/International/questions/qa-personal-names

Reply:
Unfortunately, we are dependent on an API which expects first & last name

https://docs.adyen.com/api-explorer/#/Account/v6/post/createAccountHolder__resParam_accountHolderDetails-individualDetails-name-firstName